### PR TITLE
ofs: add umd driver and hps app

### PR DESCRIPTION
**Pull request by r-rojo**
* Add ofs/umd/cpeng
  This contains ofs_cpeng.yml which contains register map and APIs
* Add ofs/apps/hps
  * This uses afu_test framework to reuse command line parsing to locate
    an AFU and give it to a command.
  * Add cpeng command that uses the ofs_cpeng driver to copy an image
    from a file.
* Add unit tests for ofs_cpeng driver

Signed-off-by: Rodrigo Rojo <rodrigo.rojo@intel.com>

### CODE REVIEW

File: ofs/apps/cpeng/hps.cpp [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  {
+    std::map<std::string, uint32_t> units = {{"s", 1E6}, {"ms", 1E3}, {"us", 1}};
+    app->add_option("-f,--filename", filename_, "Image file to copy")
+      ->default_val("hps.img")
+      ->check(CLI::ExistingFile);
+    app->add_option("-d,--destination", destination_offset_, "HPS DDR Offset")
+      ->default_val(destination_offset_);
+    app->add_option("-t,--timeout", timeout_usec_, "Timeout")
+      ->default_val(timeout_usec_)
+      ->transform(CLI::AsNumberWithUnit(units));
```
Comment (tswhison) [inline ofs/apps/cpeng/hps.cpp:75]: whoah 😃

File: ofs/apps/cpeng/hps.cpp [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  std::shared_ptr<spdlog::logger> log_;
+};
+
+int main(int argc, char *argv[])
+{
+  opae::afu_test::afu hps(cpeng_guid);
+  hps.register_command<cpeng>();
+  hps.main(argc, argv);
+  return 0;
+}
```
Comment (tswhison) [inline ofs/apps/cpeng/hps.cpp:138]: Windows or no line ending..

File: ofs/umd/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+## ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+## LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+## CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+## SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+## INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+## CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+## ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+## POSSIBILITY OF SUCH DAMAGE.
+
+opae_add_subdirectory(cpeng)
```
Comment (tswhison) [inline ofs/umd/CMakeLists.txt:27]: Windows or no line ending

**Review by michelleyho — DISMISSED** [review]

File: ofs/apps/cpeng/hps.cpp (hunk 2 above)
Comment (r-rojo) [reply ofs/apps/cpeng/hps.cpp]: fixed

**Review by tswhison — APPROVED** [review]

**Review by hundeboll-silicom — APPROVED** [review]
Apart from the generous drips of newlines: LGTM :)

**Review by michelleyho — APPROVED** [review]

**Pull request merged**